### PR TITLE
Use Myst substitution extension to populate version in citation text

### LINKED ISSUE #634: Keep version number in documentation citation up-to-date with bumpversion
This PR removed the version from the citation, but we should add it back and auto-update it:

https://github.com/nsidc/qgreenland/pull/445/files

... Maybe we don't need to do this? Should our citation be generic and version-less? What if a user used the plugin to get some QGreenland layers? I think just citing "QGreenland" without worrying about whether Core or Custom was used is best.

### PR DESCRIPTION
**Pull request by MattF-NSIDC**
## Description

title! closes #634

## Checklist

If an item on this list is done _or not needed_, check it with `[x]` or click the
checkbox.

- [x] The PR description links to issues that it resolves with `closes #{issue_number}`
- [x] Config lockfile updated (`inv config.export > qgreenland/config/cfg-lock.json`)
- [x] Environment lockfile updated if needed (`conda-lock`)
- [x] Version bumped if needed (`bumpversion (major|minor|patch|prerelease|build)`)
- [x] CHANGELOG.md updated (for user-facing changes)
- [x] Documentation updated if needed
- [x] New unit tests if needed

### CODE REVIEW

File: environments/main/environment.yml [hunk 1]
```diff
@@ -71,8 +76,4 @@ dependencies:
 
   # Pip dependencies could be imported or non-imported :(
   - pip:
-    - sphinx-rtd-theme ~=1.0.0
-    - autodoc-pydantic ~=1.5.1
     - sphinx-selective-exclude ~=1.0.3
-    # TODO: get this dependency from conda-forge (see https://github.com/callowayproject/bump-my-version/issues/46)
-    - bump-my-version ~=0.9.2
```
Comment (MattF-NSIDC) [inline environments/main/environment.yml:78]: All but one is now on conda-forge. I was hoping to get rid of this whole section!

File: doc/citing.md [hunk 2]
```diff
@@ -38,12 +38,21 @@ Please choose the appropriate citation depending on which you use.
 An asterisk (`*`) indicates co-equal lead authorship (listed alphabetically).
 
 
-### Version 1.0
+### Latest version: {{ release }}
```
Comment (MattF-NSIDC) [inline doc/citing.md:41]: I felt latest at top, descending order was more intuitive

File: doc/conf.py [hunk 3]
```diff
@@ -64,6 +64,12 @@
 # -- MyST options -------------------------------------------------
 
 myst_heading_anchors = 3
+myst_enable_extensions = [
+    "substitution",
+]
+myst_substitutions = {
+    "release": release,
```
Comment (MattF-NSIDC) [inline doc/conf.py:71]: Enables jinja syntax for substitution

File: environments/main/environment.yml [hunk 4]
```diff
@@ -41,10 +41,14 @@ dependencies:
   #############################
 
   # documentation dependencies
-  - sphinx ~=4.3.1
+  # TODO: Upgrade sphinx ones sphinx-rtd-theme supports 7
+  # - sphinx ~=7.2.2
+  - sphinx ~=6.2.1
```
Comment (MattF-NSIDC) [inline environments/main/environment.yml:46]: I was getting some weird error that just went away when I upgraded.

**Review by trey-stafford — APPROVED** [review]
Cool!

**Pull request merged**